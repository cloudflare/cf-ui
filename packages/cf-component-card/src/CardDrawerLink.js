import React from 'react';
import PropTypes from 'prop-types';
import Link from 'cf-component-link';

class CardDrawerLink extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    e.preventDefault();
    this.props.onClick();
  }

  render() {
    let className = 'cf-card__toolbar_link';

    if (this.props.isActive) {
      className += ' cf-card__toolbar_link--open';
    }

    return (
      <Link
        role="tab"
        id={this.props.id}
        className={className}
        onClick={this.handleClick}
      >
        {this.props.children}
      </Link>
    );
  }
}

CardDrawerLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  id: PropTypes.string,
  children: PropTypes.node
};

export default CardDrawerLink;
