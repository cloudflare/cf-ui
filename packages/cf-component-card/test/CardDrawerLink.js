import React from 'react';
import renderer from 'react-test-renderer';
import CardDrawerLink from '../src/CardDrawerLink';

test('should render', () => {
  const component = renderer.create(
    <CardDrawerLink id="link" isActive={false} onClick={() => {}}>
      CardDrawerLink
    </CardDrawerLink>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render with isActive', () => {
  const component = renderer.create(
    <CardDrawerLink id="link" isActive={true} onClick={() => {}}>
      CardDrawerLink
    </CardDrawerLink>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
