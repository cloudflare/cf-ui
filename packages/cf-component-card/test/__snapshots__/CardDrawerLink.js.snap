// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<a
  className="cf-link cf-card__toolbar_link"
  href="#!"
  id="link"
  onClick={[Function]}
  role="tab"
>
  CardDrawerLink
</a>
`;

exports[`should render with isActive 1`] = `
<a
  className="cf-link cf-card__toolbar_link cf-card__toolbar_link--open"
  href="#!"
  id="link"
  onClick={[Function]}
  role="tab"
>
  CardDrawerLink
</a>
`;
